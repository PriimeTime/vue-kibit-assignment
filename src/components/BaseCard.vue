<script setup lang="ts"></script>

<template>
  <div class="max-w-[40rem] rounded-xl md:shadow-xl p-10 w-full">
    <slot />
  </div>
</template>
