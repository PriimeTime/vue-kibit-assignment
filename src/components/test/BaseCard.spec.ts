import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import BaseCard from "@/components/BaseCard.vue";

describe("BaseCard", () => {
  it("renders without crashing", () => {
    const wrapper = mount(BaseCard);
    expect(wrapper.exists()).toBe(true);
  });

  it("renders slot content", () => {
    const slotContent = "This is a card!";
    const wrapper = mount(BaseCard, {
      slots: {
        default: slotContent,
      },
    });
    expect(wrapper.text()).toContain(slotContent);
  });

  it("matches snapshot", () => {
    const wrapper = mount(BaseCard, {
      slots: {
        default: "Snapshot Content",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
