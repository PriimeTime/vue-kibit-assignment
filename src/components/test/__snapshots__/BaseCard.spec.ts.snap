// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseCard > matches snapshot 1`] = `"<div class="max-w-[40rem] rounded-xl md:shadow-xl p-10 w-full">Snapshot Content</div>"`;
